### ISSUE TITLE
Unable to run single test in IntelliJ when the @DataProvider annotation has a format string

### ISSUE THREAD

**Issue by spaceCamel** [TNG/junit-dataprovider#47]
Running in IntelliJ only `testMethod` works fine, while running `testMethodFailing` triggers
`java.lang.Exception: No tests found matching Method testMethodFailing(net.spacecamel.junitdataprovider.MyTest) from org.junit.internal.requests.ClassRequest@5eb5c224`

Apparently the problem happens when the format string contains "%p[0]". it works fine if the format string contains only "%m".

```java
package net.spacecamel.junitdataprovider;

import com.tngtech.java.junit.dataprovider.DataProvider;
import com.tngtech.java.junit.dataprovider.DataProviderRunner;
import com.tngtech.java.junit.dataprovider.UseDataProvider;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.Arrays;
import java.util.List;

@RunWith(DataProviderRunner.class)
public class MyTest
{
    @Test
    @UseDataProvider(value = "data_table")
    public void testMethod(String input)
    {
        System.out.println(input);
    }

    @DataProvider
    public static List<List<Object>> data_table()
    {
        return Arrays.asList(Arrays.<Object>asList("test1"), Arrays.<Object>asList("test2"));
    }

    @Test
    @UseDataProvider(value = "data_provider_with_format_string")
    public void testMethodFailing(String input)
    {
        System.out.println(input);
    }

    @DataProvider(format = "%p[0]")
    public static List<List<Object>> data_provider_with_format_string()
    {
        return Arrays.asList(Arrays.<Object>asList("test1"), Arrays.<Object>asList("test2"));
    }
}
```

**Comment by aaschmid** [comment]
Hi @spaceCamel,

thanks for reporting this. Unfortunately, this behavior is quite obvious because the filtering of test methods is done via its ```Description```. By providing a ```format``` the ```Description``` is changed to the desired format. This is documented in more detail at  https://github.com/TNG/junit-dataprovider/wiki/Features#customize-test-method-name:

> But be aware of using this feature, as it may break the following other features:

> * Running all tests for one dataprovider (in various IDEs) and jumping to a test method by double-clicking it (in Eclipse) does only work if the custom @DataProvider#format() starts with the test method name (= %m) followed by a character which is not a valid java identifier, e.g. " ", "[", "(", ":", or "-".
> * If test method names are not unique within its class, it can cause indeterministic behavior!

You could use something like ```%m - %p[0]``` instead. I can also think about a possible fix but therefore I would like to see a good use case, what is your use case?

Cheers,
Andreas

**Comment by aaschmid** [comment]
The bug described above is now fixed with commit 4c7f43f. Can you please describe your use case such that I can decide if it is worth to fix the filtering of tests for formats like ```%p[0]```?

**Comment by aaschmid** [comment]
@spaceCamel: please refer to #48 for additional discussions. The release [1.10.0](https://github.com/TNG/junit-dataprovider/releases/tag/v1.10.0) solves at least the problem of filtering for tests which start with test method name!
